gtsfm/configs/synthetic_front_end.yaml

```diff
@@ -0,0 +1,102 @@
+# Synthetic front-end configuration specifically for the Tanks & Temples dataset.
+
+SceneOptimizer:
+  _target_: gtsfm.scene_optimizer.SceneOptimizer
+  save_gtsfm_data: True
+  save_two_view_correspondences_viz: False
+  save_3d_viz: True
+  pose_angular_error_thresh: 5 # degrees
+
+  image_pairs_generator:
+    _target_: gtsfm.retriever.image_pairs_generator.ImagePairsGenerator
+    global_descriptor:
+      _target_: gtsfm.frontend.cacher.global_descriptor_cacher.GlobalDescriptorCacher
+      global_descriptor_obj:
+        _target_: gtsfm.frontend.global_descriptor.netvlad_global_descriptor.NetVLADGlobalDescriptor
+    retriever:
+      _target_: gtsfm.retriever.joint_netvlad_sequential_retriever.JointNetVLADSequentialRetriever
+      num_matched: 2
+      min_score: 0.2
+      max_frame_lookahead: 3
+
+    # retriever:
+    #   _target_: gtsfm.retriever.sequential_retriever.SequentialRetriever
+    #   max_frame_lookahead: 4
+
+    # retriever:
+    #   _target_: gtsfm.retriever.netvlad_retriever.NetVLADRetriever
+    #   num_matched: 50
+    #   min_score: 0.3
+
+  correspondence_generator:
+    _target_: gtsfm.frontend.correspondence_generator.synthetic_correspondence_generator.SyntheticCorrespondenceGenerator
+    #dataset_root: /Users/johnlambert/Downloads/Tanks_and_Temples_Barn_410
+    #dataset_root: /usr/local/gtsfm-data/Tanks_and_Temples_Barn_410
+    dataset_root: /home/runner/work/gtsfm/gtsfm/Tanks_and_Temples_Barn_410 # Path for CI.
```

Comment:
I'd prefer this being loaded from the dataset root passed from the terminal, but its not an easy fix. So we should be fine with it.